src/transaction/log_tran_table.c

```diff
@@ -1473,6 +1475,46 @@ logtb_free_tran_mvcc_info (LOG_TDES * tdes)
   curr_mvcc_info->sub_ids.clear ();
 }
 
+/*
+ * logtb_set_session_tdes - reflect the session or system parameters to the transaction
+ *
+ * return: nothing
+ *
+ *   tdes(in/out): Transaction descriptor
+ */
+static void
+logtb_set_session_tdes (THREAD_ENTRY * thread_p, LOG_TDES * tdes)
+{
+#if defined (SERVER_MODE)
+  SESSION_PARAM *session_parameter;
+#endif
+  TRAN_ISOLATION isolation;
+  int wait_msecs;
+
+#if defined (SERVER_MODE)
+  session_parameter = session_get_session_parameter (thread_p, PRM_ID_LOG_ISOLATION_LEVEL);
+  assert (session_parameter != NULL);
+  isolation = (TRAN_ISOLATION) session_parameter->value.i;
+
+  session_parameter = session_get_session_parameter (thread_p, PRM_ID_LK_TIMEOUT);
+  assert (session_parameter != NULL);
+  wait_msecs = session_parameter->value.i;
+#else /* SERVER_MODE */
```

Comment:
in `prm_get_value()`:

```
11123   if (PRM_SERVER_SESSION (prm_id) && BO_IS_SERVER_RESTARTED ())
11124     {
11125       SESSION_PARAM *sprm;
11126       thread_p = thread_get_thread_entry_info ();
11127       sprm = session_get_session_parameter (thread_p, prm_id);
11128       if (sprm)
11129         {
11130           return &(sprm->value);
11131         }
11132     }
```

Reply:
I misunderstood because `prm_get_integer_value ()` referenced `prm_Def`. But it's all handled internally. Sorry.
The code becomes as simple as below.

```
/* isolation level */
tdes->isolation = (TRAN_ISOLATION) prm_get_integer_value (PRM_ID_LOG_ISOLATION_LEVEL);

/* lock timeout */
tdes->wait_msecs = prm_get_integer_value (PRM_ID_LK_TIMEOUT);
if (tdes->wait_msecs > 0)
{
  tdes->wait_msecs = tdes->wait_msecs * 1000;
}
```

Then, I'm thinking about the wrapping function `logtb_set_session_tdes ()` which is detached, because it reflects the session.
Is function `logtb_set_session_tdes ()` necessary? What do you think.

Reply:
It's good! I'll fix it now.